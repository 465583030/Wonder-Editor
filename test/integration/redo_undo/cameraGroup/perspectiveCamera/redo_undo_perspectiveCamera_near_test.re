open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: perspectiveCamera near", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => sandbox := createSandbox());
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    let _changeNear = value => {
      let nearDomIndex = MainEditorCameraProjectionTool.getNearDomIndex();

      MainEditorCameraProjectionTool.triggerPerspectiveCameraChangeAndBlurEvent(
        nearDomIndex,
        value,
      );
    };

    let _simulateTwiceChangeNear = () => {
      let value1 = 10.112;
      let value2 = 123.12;

      _changeNear(value1);
      _changeNear(value2);
    };

    let _beforeEach = () => {
      MainEditorSceneTool.initStateWithJob(
        ~sandbox,
        ~noWorkerJobRecord=
          NoWorkerJobConfigToolEngine.buildNoWorkerJobConfig(),
        (),
      );

      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;

      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorSceneTool.setFirstCameraTobeCurrentSceneTreeNode,
      );
      DirectorToolEngine.prepareAndInitAllEnginState();
    };

    RedoUndoTool.testRedoUndoTwoStep(
      sandbox,
      "prepare first step: set currentSceneTreeNode",
      (_simulateTwiceChangeNear, _beforeEach, () => ()),
      BuildComponentForRedoUndoTool.buildCameraProjection,
    );
  });