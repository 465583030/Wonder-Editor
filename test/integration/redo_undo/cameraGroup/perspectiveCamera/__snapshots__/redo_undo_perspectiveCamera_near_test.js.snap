// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: perspectiveCamera near prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-camera-view"
>
  <article
    className="wonder-select"
  >
    <span
      className=""
    >
      type : 
    </span>
    <select
      onChange={[Function]}
      value="0"
    >
      <option
        value="0"
      >
        perspective
      </option>
    </select>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Near : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="10.112"
      />
    </article>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Far : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1000"
      />
    </article>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Fovy : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="60"
      />
    </article>
  </article>
</article>
`;

exports[`redo_undo: perspectiveCamera near prepare first step: set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<article
  className="wonder-camera-view"
>
  <article
    className="wonder-select"
  >
    <span
      className=""
    >
      type : 
    </span>
    <select
      onChange={[Function]}
      value="0"
    >
      <option
        value="0"
      >
        perspective
      </option>
    </select>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Near : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="123.12"
      />
    </article>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Far : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1000"
      />
    </article>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Fovy : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="60"
      />
    </article>
  </article>
</article>
`;

exports[`redo_undo: perspectiveCamera near prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-camera-view"
>
  <article
    className="wonder-select"
  >
    <span
      className=""
    >
      type : 
    </span>
    <select
      onChange={[Function]}
      value="0"
    >
      <option
        value="0"
      >
        perspective
      </option>
    </select>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Near : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="123.12"
      />
    </article>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Far : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1000"
      />
    </article>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Fovy : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="60"
      />
    </article>
  </article>
</article>
`;

exports[`redo_undo: perspectiveCamera near prepare first step: set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-camera-view"
>
  <article
    className="wonder-select"
  >
    <span
      className=""
    >
      type : 
    </span>
    <select
      onChange={[Function]}
      value="0"
    >
      <option
        value="0"
      >
        perspective
      </option>
    </select>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Near : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="10.112"
      />
    </article>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Far : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1000"
      />
    </article>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Fovy : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="60"
      />
    </article>
  </article>
</article>
`;

exports[`redo_undo: perspectiveCamera near prepare first step: set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-camera-view"
>
  <article
    className="wonder-select"
  >
    <span
      className=""
    >
      type : 
    </span>
    <select
      onChange={[Function]}
      value="0"
    >
      <option
        value="0"
      >
        perspective
      </option>
    </select>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Near : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.1"
      />
    </article>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Far : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1000"
      />
    </article>
  </article>
  <article
    className="wonder-floatInput-base"
  >
    <article
      className="wonder-float-input"
    >
      <span
        className="component-label"
      >
        Fovy : 
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="60"
      />
    </article>
  </article>
</article>
`;
